### DISCUSSION TITLE
[Feature]: Training and testing splitting.

### DISCUSSION THREAD

**Discussion by VittoriaOssanna** [MilesCranmer/PySR#778] [Q&A] [answered]
### Feature Request

Good morning everyone, 

I am new to PySR, so apologies if the questions I place are naiive.
I was wondering if there is any way to have an automatic "training and testing" split, and for training I mean the value for fitting, while for testing unseen data on which the loss can be calculated. 

I am trying to implement this feature for my project, and I see that `model.predict()` evaluates the input based on the equation chosen by `model_selection`. Is there any way of evaluating the expression also on other candidate functions?

Thank you for your help!

**Comment by MilesCranmer** [comment, marked as answer]
You can use scikit learn’s train test split: https://scikit-learn.org/1.5/modules/generated/sklearn.model_selection.train_test_split.html, and then feed in the train split to `.fit` and the test split to `.predict`

Note that `.predict` also has a keyword for the equation index so you can test with other ones too. See the docs for more info: https://ai.damtp.cam.ac.uk/pysr/examples/

hope this helps!
Best,
Miles

**Comment by VittoriaOssanna** [comment]
Thank you for your answer!
I was also wondering at this point if there is a default method to calculate the loss on the test set, instead of predict. 

Thank you again!
